typescript/testsuite/testsuite_impl/basic_datastore_and_api_test/basic_datastore_and_api_test_.ts

```diff
@@ -0,0 +1,238 @@
+		return ok(networkCtx);
+	}
+
+	public async run(network: Network): Promise<Result<null, Error>> { //TODO(Ali) - async?
+		// Go doesn't have generics so we have to do this cast first
+		const castedNetwork: NetworkContext = <NetworkContext>network;
+
+		const serviceContextResult: Result<ServiceContext, Error> = await castedNetwork.getServiceContext(API_SERVICE_ID);
+		if (!serviceContextResult.isOk()) {
+			return err(serviceContextResult.error);
+		}
+		const serviceContext: ServiceContext = serviceContextResult.value;
+
+		const apiClient: APIClient = new APIClient(serviceContext.getIPAddress(), API_SERVICE_PORT);
+
+		log.info("Verifying that person with test ID '" + TEST_PERSON_ID + "' doesn't already exist...");
+		const getPersonExistsResult: Result<Person, Error> = await apiClient.getPerson(TEST_PERSON_ID);
+		if (!getPersonExistsResult.isOk()) {
+			return err(new Error("Expected an error trying to get a person who doesn't exist yet, but didn't receive one"));
+		}
+		log.info("Verified that test person doesn't already exist");
+
+		log.info("Adding test person with ID '" + TEST_PERSON_ID + "'...");
+		const addPersonResult: Result<null, Error> = await apiClient.addPerson(TEST_PERSON_ID);
+		if (!addPersonResult.isOk()) {
+			return err(addPersonResult.error);
+		}
+		log.info("Test person added");
+
+		log.info("Incrementing test person's number of books read by " + TEST_NUM_BOOKS_READ + "...");
+		for (let i = 0; i < TEST_NUM_BOOKS_READ; i++) {
+			const incrementBooksReadResult: Result<null, Error> = await apiClient.incrementBooksRead(TEST_PERSON_ID);
+			if (!incrementBooksReadResult.isOk()) {
+				return err(incrementBooksReadResult.error);
+			}
+		}
+		log.info("Incremented number of books read");
+
+		log.info("Retrieving test person to verify number of books read...");
+		const getPersonResult: Result<Person, Error> = await apiClient.getPerson(TEST_PERSON_ID);
+		if (!getPersonResult.isOk()) {
+			return err(getPersonResult.error);
+		}
+		const person: Person = getPersonResult.value;
+		log.info("Retrieved test person");
+
+		if (person.getBookRead() !== TEST_NUM_BOOKS_READ) {
+			return err(new Error("Expected number of book read '"+TEST_NUM_BOOKS_READ+"' !== actual number of books read '"+person.getBookRead()+"'"));
+		}
+
+		return ok(null);
+	}
+}
+
+// ====================================================================================================
+//                                       Private helper functions
+// ====================================================================================================
+
+//TODO TODO TODO (Ali) - review these helper methods after making final changes to network_impl
+async function getDatastoreServiceConfigurations(): Promise<[ContainerCreationConfig, (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => Result<ContainerRunConfig, Error>]> {
```

Comment:
Same comment as above - this function, IMO, actually doesn't need to exist - it's just two lines and the cost of unpacking all the results is higher than actually just inlining the lines